ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -251,10 +249,10 @@ private void clean(CompactionInfo ci, long minOpenTxnGLB, boolean metricsEnabled
 
       if (t != null) {
         StorageDescriptor sd = resolveStorageDescriptor(t, p);
-        cleanUpTask = () -> removeFiles(location.orElse(sd.getLocation()), minOpenTxnGLB, ci,
+        cleanUpTask = () -> removeFiles(sd.getLocation(), minOpenTxnGLB, ci,
```

Comment:
that doesn't look like a compatible change

Reply:
If you check line 214-218, _t_ can have value only if _location_ is null, therefore the orElse was unnecessary, and the change is compatible.